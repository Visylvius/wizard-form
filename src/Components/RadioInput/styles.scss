@import '../../variables';

.RadioInput-container {
  display: flex;
  align-items: center;

  .RadioInput-label {
    color: $radio-label;
    font-size: 12px;
    padding-left: 4px;
  }

  input[type='radio'] {
    cursor: pointer;
  }
}
