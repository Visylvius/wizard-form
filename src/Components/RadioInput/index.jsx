import React from 'react';
import classNames from 'classnames';

import './styles.scss';

const RadioInput = ({
  containerClasses,
  name,
  radioInputClasses,
  value,
  shouldShowLabel,
  labelText,
  id,
  ...rest
}) => {
  const containerClassNames = classNames(
    'RadioInput-container',
    containerClasses,
  );
  const radioInputClassNames = classNames(
    'RadioInput-component',
    radioInputClasses,
  );

  return (
    <div className={containerClassNames}>
      <input
        className={radioInputClassNames}
        type='radio'
        value={value}
        name={name}
        id={id}
        {...rest}
      />
      {shouldShowLabel && labelText && (
        <label className='RadioInput-label' htmlFor={id}>
          {labelText}
        </label>
      )}
    </div>
  );
};

export default RadioInput;
