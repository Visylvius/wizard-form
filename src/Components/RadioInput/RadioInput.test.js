import React from 'react';
import renderer from 'react-test-renderer';

import RadioInput from './index';

describe('render', () => {
  test('should render properly', () => {
    const tree = renderer.create(
      <RadioInput shouldShowLabel labelText='stuff' />,
    );

    expect(tree).toMatchSnapshot();
  });
});
