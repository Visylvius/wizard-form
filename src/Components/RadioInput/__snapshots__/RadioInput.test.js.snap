// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render properly 1`] = `
<div
  className="RadioInput-container"
>
  <input
    className="RadioInput-component"
    type="radio"
  />
  <label
    className="RadioInput-label"
  >
    stuff
  </label>
</div>
`;
